src/anemoi/inference/outputs/gribfile.py

```diff
@@ -169,16 +177,16 @@ def __init__(
             write_initial_state=write_initial_state,
             variables=variables,
         )
-        self.path = path
-        self.output = GribWriter(self.path, split_output=split_output)
+        self.out = out
+        self.output = GribWriter(self.out, split_output)
         self.archiving = defaultdict(ArchiveCollector)
         self.archive_requests = archive_requests
         self.check_encoding = check_encoding
         self._namespace_bug_fix = False
 
     def __repr__(self) -> str:
-        """Return a string representation of the GribFileOutput object."""
-        return f"GribFileOutput({self.path})"
+        """Return a string representation of the GribOutput object."""
+        return f"GribOutput({self.out})"
```

Comment:
Instead of hardcoding the class name, you could just do something like `type(self).__name__` ? Then the derived classes will automatically print their own names.